### DISCUSSION TITLE
Why is Watchower not recommended for use in a production environment?

### DISCUSSION THREAD

**Discussion by JhnW** [containrrr/watchtower#1890] [Q&A] [answered]
I understand the potential information about the direction of the application's development, it's ok for me. I was just considering using Watchower as a medium-sized, non-commercial discussion forum (technical migration, the new server we are targeting will have 2 GB of RAM, plus the engine and database requirements , memory footprint of alternatives is too large + this is overkill).
Are there any additional messages that I don't know about, that are implicitly embedded in the essence of the project that would be a strong argument against this type of use?

**Comment by piksel** [comment, marked as answer]
The main problem with using watchtower is that due to how it works, it may leave you with no running container and no easy way to recreate it if it fails to do an upgrade. We try to make this as unlikely to happen as possible, but it cannot be guaranteed to always work. 

When relying on an application ("in production"), it's generally a better idea to explicitly do the upgrades at fixed times with some (human) monitoring to avoid problems causing extended downtime.

The main point is that doing this type of upgrade is a "hack" (since why it's not supported by docker itself), and doing it the docker intended way is safer when possible. That's why we cannot recommend using it in production.

**Comment by Eirikr70** [comment]
In production, as far as I am concerned, I would recommend using Watchtower in monitor-only mode. Then you get notified that a new release is available, you have it pulled, but it's still up to you to upgrade.
